statsmodels/genmod/generalized_linear_model.py

```diff
@@ -749,21 +762,21 @@ def fit(self, start_params=None, maxiter=100, method='IRLS', tol=1e-8,
         -----
         This method does not take any extra undocumented ``kwargs``.
         """
-
-        endog = self.endog
-        if endog.ndim > 1 and endog.shape[1] == 2:
-            data_weights = endog.sum(1)  # weights are total trials
-        else:
-            data_weights = np.ones((endog.shape[0]))
-        self.data_weights = data_weights
-        if np.shape(self.data_weights) == () and self.data_weights > 1:
-            self.data_weights = self.data_weights * np.ones((endog.shape[0]))
-        self.scaletype = scale
-        if isinstance(self.family, families.Binomial):
         # this checks what kind of data is given for Binomial.
         # family will need a reference to endog if this is to be removed from
         # preprocessing
-            self.endog = self.family.initialize(self.endog)
+        self.n = np.ones((self.endog.shape[0]))  # For binomial
+        if isinstance(self.family, families.Binomial):
+            tmp = self.family.initialize(self.endog)
+            self.endog = tmp[0]
+            self.n = tmp[1]
+        if self.freq_weights is None:
+            self.freq_weights = np.ones((self.endog.shape[0]))
+        if np.shape(self.freq_weights) == () and self.freq_weights > 1:
+            self.freq_weights = (self.freq_weights *
+                                 np.ones((self.endog.shape[0])))
+        self.freq_weights = self.n * self.freq_weights
```

Comment:
Given that self.n is already the correct length array, this works also if freq_weights is scalar. So previous if looks redundant. Either n or freq_weights can be initialized to scalar 1.